File: javascript/test/mf.js
#!/usr/bin/env node
const liba = require("../liba");
function assert(expr) {
    if (!eval(expr)) {
        throw Error(expr);
    }
}
assert("liba.mf.NUL == 0");
assert("liba.mf.GAUSS == 1");
assert("liba.mf.GAUSS2 == 2");
assert("liba.mf.GBELL == 3");
assert("liba.mf.SIG == 4");
assert("liba.mf.DSIG == 5");
assert("liba.mf.PSIG == 6");
assert("liba.mf.TRAP == 7");
assert("liba.mf.TRI == 8");
assert("liba.mf.LINS == 9");
assert("liba.mf.LINZ == 10");
assert("liba.mf.S == 11");
assert("liba.mf.Z == 12");
assert("liba.mf.PI == 13");
for (var i = -1; i < 1; i += 0.1) {
    console.log(
        liba.mf.gauss(i, 1, 0),
        liba.mf.gauss2(i, 1, -1, 1, 1),
        liba.mf.gbell(i, 2, 1, 0),
        liba.mf.sig(i, 2, 0),
        liba.mf.dsig(i, 5, -2, 5, 2),
        liba.mf.psig(i, 5, -2, -5, 2),
        liba.mf.trap(i, -2, -1, 1, 2),
        liba.mf.tri(i, -1, 0, 1),
        liba.mf.lins(i, -1, 1),
        liba.mf.linz(i, -1, 1),
        liba.mf.s(i, -1, 1),
        liba.mf.z(i, -1, 1),
        liba.mf.pi(i, -2, -1, 1, 2)
    );
}

File: python/test/mf.py
#!/usr/bin/env python
from __future__ import print_function
import os, sys

path = os.path.dirname(__file__)
path = os.path.dirname(path)
sys.path.insert(0, path)
if len(sys.argv) > 1:
    sys.stdout = open(sys.argv[1], "w")

import liba  # type: ignore

N = 1000
T = 6 / N
x = liba.new_real(N)
for i in range(N):
    x[i] = (i - N / 2) * T

liba.mf.gauss(x, 1, 0)
liba.mf.gauss2(x, 1, -1, 1, +1)
liba.mf.gbell(x, 2, 4, 0)
liba.mf.sig(x, 2, 0)
liba.mf.dsig(x, 5, -2, +5, 2)
liba.mf.psig(x, 5, -2, -5, 2)
liba.mf.tri(x, 0, 1, 2)
liba.mf.trap(x, 0, 1, 2, 3)
liba.mf.lins(x, 1, 2)
liba.mf.linz(x, 1, 2)
liba.mf.s(x, 1, 2)
liba.mf.z(x, 1, 2)
liba.mf.pi(x, 0, 1, 2, 3)

S = 1
params = [
    liba.new_real((liba.mf.TRI, -3 * S, -3 * S, -2 * S)),
    liba.new_real((liba.mf.TRI, -3 * S, -2 * S, -1 * S)),
    liba.new_real((liba.mf.TRI, -2 * S, -1 * S, +0 * S)),
    liba.new_real((liba.mf.TRI, -1 * S, +0 * S, +1 * S)),
    liba.new_real((liba.mf.TRI, +0 * S, +1 * S, +2 * S)),
    liba.new_real((liba.mf.TRI, +1 * S, +2 * S, +3 * S)),
    liba.new_real((liba.mf.TRI, +2 * S, +3 * S, +3 * S)),
]
for param in params:
    liba.mf.mf(param[0], x, param[1:])
